_posts/2021-08-11-persistent-identifiers.md

```diff
@@ -0,0 +1,84 @@
+
+## How are we converting our internal identifiers to URIs?
+
+For each dataset we define a sub-domain under ``http://data.bgs.ac.uk/id``
+
+e.g. ``http://data.bgs.ac.uk/id/dataHolding/``  for our data catalogue (metadata) entries
+
+Sometimes the url may contain another part in the path to group identifiers that belong to a larger concept scheme 
+
+e.g.
+``http://data.bgs.ac.uk/id/Lexicon/NamedRockUnit/``  for our rock formations described in the [BGS Lexicon of Named Rock Units](https://www.bgs.ac.uk/technologies/the-bgs-lexicon-of-named-rock-units/)
+
+We append the integer or text code from each item in the database to the appropriate uri subdomain stem
+
+e.g. 
+``http://data.bgs.ac.uk/id/dataHolding/13606914``
+
+``http://data.bgs.ac.uk/id/Lexicon/NamedRockUnit/OXC``
+
+Here is where we had a problem in that some of these datasets pre-date the World Wide Web and Linked Data, and a small proportion of the text identifiers contained characters that are illegal and/or problematic when included in a url. The codes are widely used in internal database references and used in key datasets such as our digital geological mapping and borehole logging so we couldn't update them at source without having to do a lot of unpicking. To solve this we created and used a secondary unique key on these database tables where the illegal characters were substituted with a legal character (e.g. ``H`` instead of ``#``) using an easily reversible algorithm, making sure we maintained uniqueness.
+
+## How are we exposing these on the web?
+
+
+We run nightly scripts through a gitlab scheduled ci/cd pipeline that query our relational database and generate linked data triples declaring each of these uris as types of [skos:Concept](https://www.w3.org/2009/08/skos-reference/skos.html#Concept)
+
+[The scripts actually construct lots of other triples to serialise other information that we know about those concepts too - that's another story.]
+
+For some datasets we have a separate application that provides the landing page and search functionality for that dataset (for example we use Geonetwork for our data catalogue). In these cases we also construct a triple that associates the concept with the url of its landing page using the [foaf:homepage](http://xmlns.com/foaf/spec/#term_homepage) predicate.
+
+As part of the scheduled pipeline, all the triples are updated nightly to https://github.com/BritishGeologicalSurvey/vocabularies
+and the triples are re-loaded into our Apache Jena triplestore.  
+
+The linked data is exposed to the web using nginx and a Linked Data API written using the python based Falcon web framework. 
+
+## How do we redirect to a user friendly web page or application?
+
+
+The API handles content negotiation for different RDF (resource description framework) formats and uses [rdflib](https://github.com/RDFLib/rdflib) for serialisation. 
```

Comment:
I got a little lost here.  I'm not quite sure why http://data.bgs.ac.uk/id/dataHolding/13606914.nt doesn't redirect but http://data.bgs.ac.uk/id/dataHolding/13606914 does redirect. Can you just read over again? I'm happy for you to conclude it's clear!

Reply:
Good point - not easy to spot the difference - I've added some clarification